src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamWatcher.java

```diff
@@ -189,7 +189,8 @@ private synchronized void onInitialImageStream(ImageStreamList imageStreams) {
             }
             // for IST, can't set labels, so check annotations
             if (ist != null && hasSlaveLabelOrAnnotation(ist.getMetadata().getAnnotations())) {
-                results.add(this.podTemplateFromData(ist.getMetadata().getName(),
+                // note, pod names cannot have colons
+                results.add(this.podTemplateFromData(ist.getMetadata().getName().replaceAll(":", "."),
```

Comment:
so now "my.imagestream:foo" and "my:imagestream.foo" are going to get mapped to the same thing.  are dots allowed in imagestream names and tag names? i suspect they are.

unfortunately this is hard problem for any separator unless there's a value that's valid in pod names and not valid in imagestreams or tag names.